File: 2/calculadora.py
print("______________CALCULADORA________________\n1 Suma\n2 Resta\n3 Multiplicacion\n4 Division")
op = int(input("Elija una Opcion 1-4: "))
a = int(input("Introduzca el 1er numero: "))
b = int(input("Introduzca el 2do numero: "))
res = 0
suma = lambda a,b:a+b
resta = lambda a,b:a-b
mult = lambda a,b:a*b
div = lambda a,b:a/b
if (op == 1):
    res = suma(a,b)
elif (op == 2):
    res = resta(a,b)  
elif (op == 3):
    res = mult(a,b)  
elif (op == 4):
    res = div(a,b)
else:
    res = "Opcion invalida"
print(f'El resultado es: {res}')  

File: 2/calculadora.scala
object Calculadora {
  def main(args: Array[String]): Unit = {
    print("\n______________CALCULADORA________________\nElije una opcion:\n  1 Suma\n  2 Resta\n  3 Multiplicacion\n  4 Division\n--> ")
    val op = scala.io.StdIn.readLine.toInt
    print("Introduce 1er Num: ")
    val a : Int = scala.io.StdIn.readLine.toInt
    print("Introduce 2do Num: ")
    val b : Int = scala.io.StdIn.readLine.toInt
    val suma: (Int, Int) => Int = (x: Int, y: Int) => x + y
    val resta: (Int, Int) => Int = (x: Int, y: Int) => x + y
    val mult: (Int, Int) => Int = (x: Int, y: Int) => x + y
    val div: (Int, Int) => Int = (x: Int, y: Int) => x + y
    val res: Int = if (op == 1) suma(a,b) else if (op == 2) resta(a,b) else if (op == 3) mult(a,b) else if (op == 4) div(a,b) else 0
    print("         Resultado: "+res)
  }
}